src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1481,6 +1489,42 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
 
         return null;
     }
+    
+    /**
+     * Add log event for request failure with a cause. 
+     */
+    private void addLogEventForRequestFailureWithCause(final String eventName, final AuthenticationRequest request, 
+            final AuthenticationException authenticationException) {
+        ClientAnalytics.logEvent(
```

Comment:
create method with just 1 method inside? what is the optimization?
And this method doesn't fit in 1 string anyway